tools/building.py

```diff
@@ -1481,7 +1518,7 @@ def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdou
   cmd += get_binaryen_feature_flags()
   # if we are emitting a source map, every time we load and save the wasm
   # we must tell binaryen to update it
-  if settings.GENERATE_SOURCE_MAP and outfile:
+  if settings.GENERATE_SOURCE_MAP and outfile and tool in ['wasm-opt', 'wasm-emscripten-finalize']:
```

Comment:
Why this?

Reply:
Only those tools support source maps. `wasm-ctor-eval` for example does not. I'll add a TODO